src/ESLintLinter.php

```diff
@@ -174,8 +174,7 @@ protected function parseLinterOutput($path, $err, $stdout, $stderr) {
         $message->setDescription($offense['message']);
         $message->setLine($offense['line']);
         $message->setChar($offense['column']);
-        $source = array_key_exists('source', $offense) ? $offense['source'] : $file['filePath'];
-        $message->setCode($source);
+        $message->setCode($this->getLinterName());
```

Comment:
What do you think about using `ruleId` here when it's available? Having a unique code per error type is useful because it allows the user to configure (and disable) individual severities.

```php
$message->setCode($offense['ruleId'] || $this->getLinterName());
```

And above, we could prefix the name with the linter's name, too:

```php
$message->setName($this->getLinterName() . ($offense['ruleId'] || ''));
```

Reply:
> Having a unique code per error type is useful because it allows the user to configure (and disable) individual severities.

IMHO the idiomatic place to do that mapping is the `.eslintrc.yml` - that way other tools (CI/IDE etc) all use the same values.

But doesn't seem like a bad idea to support it anyway. But that seems like a separate task?

> And above, we could prefix the name with the linter's name, too:

I would probably not prefix it to the name - I think arc lint print's that name separately (uses the name you specified in `.arclint` though).

I only used the linter name for `setCode` because that's what most other linters do when they only forward lint messages from an external tool - see the included [go linter for example](https://github.com/phacility/arcanist/blob/master/src/lint/linter/ArcanistGoLintLinter.php#L55).

Reply:
The [flake8 linter](https://github.com/phacility/arcanist/blob/master/src/lint/linter/ArcanistFlake8Linter.php#L75-L78) is an alternate example where the code _is_ forwarded on through the ArcanistLinter interface.

... but that code reference also reminds me that my comment about customizing severities is more about calling `$message->setSeverity()` rather than `$message->setCode()`.